File: Assets/Scripts/Core/GameOverResult.cs
using System.Collections.Generic;

namespace Bunker.Core
{
    // Describes why and how the game ended, so UI can show an appropriate
    // message instead of just "you won" for every case.
    public class GameOverResult
    {
        public enum Reason
        {
            SurvivorsTargetReached,   // normal win: exactly SurvivorsTarget players remain
            AllPlayersEliminated      // edge case: ties/rules led to zero survivors
        }

        public Reason EndReason;
        public List<PlayerData> Survivors;
    }
}

File: bunker-server/src/game/GameOverResult.ts
import { PlayerData } from "./PlayerData";

// Equivalent of the C# GameOverResult (section 1.7).
export enum GameOverReason {
  SurvivorsTargetReached = 0,
  AllPlayersEliminated = 1,
}

export interface GameOverResult {
  endReason: GameOverReason;
  survivors: PlayerData[];
}
